packages/qwik/src/core/shared/shared-serialization.ts

```diff
@@ -760,8 +758,16 @@ export const createSerializationContext = (
       } else if (obj instanceof Error) {
         discoveredValues.push(...Object.values(obj));
       } else if (isStore(obj)) {
-        discoveredValues.push(getStoreTarget(obj));
-        discoveredValues.push(getStoreHandler(obj)!.$effects$);
+        const target = getStoreTarget(obj)!;
+        const effects = getStoreHandler(obj)!.$effects$;
+        discoveredValues.push(target, effects);
```

Comment:
Small optimization: the effects will always be a unique array so skip it

```suggestion
        discoveredValues.push(target, ...effects);
```